server.js

```diff
@@ -20,13 +15,95 @@ mongoose.Promise = Promise;
 const port = process.env.PORT || 8080;
 const app = express();
 
+//Seeding with database
+
+const Song = mongoose.model("Song", {
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number,
+});
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Song.deleteMany({});
+
+    topMusicData.forEach((songData) => {
+      new Song(songData).save();
+    });
+  };
+
+  seedDatabase();
+}
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(expressListEndpoints(app));
+});
+
+app.get("/songs", async (req, res) => {
+  const songs = await Song.find();
+  res.json(songs);
+});
+
+app.get("/songs/:id", async (req, res) => {
+  const songID = req.params.id;
+  const songInfo = await Song.findOne({ id: songID }).exec();
+  if (songInfo !== null) {
+    res.json(songInfo);
+  } else {
+    res.status(404).send("Information not found, please try again");
+  }
+});
+
+app.get("/songs/genre/:genre", async (req, res) => {
+  const songGenre = req.params.genre;
+  const songsByGenre = await Song.find({
+    genre: songGenre,
+  }).exec();
+  if (songsByGenre !== null) {
+    res.json(songsByGenre);
+  } else {
+    res.status(404).send("Information not found, please try again");
+  }
+});
+
+app.get("/songs/popularity/:popularityScore", async (req, res) => {
+  const songPopularity = req.params.popularityScore;
+  const songsByPopularity = await Song.find({
+    popularity: { $gte: songPopularity },
+  }).exec();
+  if (songsByPopularity !== null) {
+    res.json(songsByPopularity);
+  } else {
+    res.status(404).send("Information not found, please try again");
+  }
+});
+
+app.get("/songs/artist/:artistName", async (req, res) => {
+  const artistName = req.params.artistName;
+  const songsByArtist = await Song.find({
+    artistName,
+  }).exec();
+  if (songsByArtist !== null) {
+    res.json(songsByArtist);
+  } else {
+    res.status(404).send("Information not found, please try again");
```

Comment:
Same thing as I wrote in the beginning, could be helpful to guide the user on how to search as it's not evident. For example I noticed that it is case sensitive so that could be a good thing to write to help the user :)

Reply:
Nice point Alma. The current implementation of the /songs/artist/:artistName endpoint will perform an exact match, which is case-sensitive. To include lowercase and make the search case-insensitive, you can use a regular expression with the $regex operator and the $options set to "i" (for case-insensitive). For example:

```javascript
app.get("/songs/artist/:artistName", async (req, res) => {
  try {
    const artistName = req.params.artistName;
    const songsByArtist = await Song.find({
      artistName: { $regex: new RegExp(artistName, "i") },
    }).exec();

    if (songsByArtist.length > 0) {
      res.json(songsByArtist);
    } else {
      res.status(404).json({ error: "No songs found for this artist" });
    }
  } catch (error) {
    res.status(500).json({ error: "Error fetching songs by artist" });
  }
});
```